File: dev-portal/python/postman_sdk/types/invitation.py
"""
Partner invitation-related types
"""

from typing import Literal
from pydantic import BaseModel, Field


InvitationStatus = Literal[
    "EMAIL_SENT",
    "PARTNER_ADDED",
    "PENDING",
    "ACCEPTED",
    "EXPIRED",
    "FAILED",
]


class InvitePartnerRequest(BaseModel):
    """Partner invitation request"""
    workspace_id: str
    email: str
    role_id: str | None = None


class InvitePartnerResult(BaseModel):
    """Partner invitation result"""
    success: bool
    email: str
    status: InvitationStatus | None = None
    invitation_link: str | None = None
    user_id: str | None = None
    role_display_name: str | None = None
    error: str | None = None


class RemovePartnerRequest(BaseModel):
    """Remove partner request"""
    workspace_id: str
    user_id: str


class RemovePartnerResult(BaseModel):
    """Remove partner result"""
    success: bool
    user_id: str
    status: str | None = None
    error: str | None = None


class InvitationData(BaseModel):
    """Invitation data stored in memory"""
    status: InvitationStatus
    invitation_link: str | None = None
    user_id: str | None = None
    role_display_name: str | None = None


class InvitationLink(BaseModel):
    """Invitation link info"""
    email: str
    invitation_link: str
    status: InvitationStatus


class BatchInviteResultItem(BaseModel):
    """Batch invite failed item"""
    email: str
    error: str


class BatchInviteResult(BaseModel):
    """Batch invite partners result"""
    success: list[InvitePartnerResult] = Field(default_factory=list)
    failed: list[BatchInviteResultItem] = Field(default_factory=list)

File: dev-portal/typescript/sdk/src/types/invitation.ts
/**
 * Partner invitation-related types
 */

/**
 * Partner invitation status
 */
export type InvitationStatus = 
  | 'EMAIL_SENT'
  | 'PARTNER_ADDED'
  | 'PENDING'
  | 'ACCEPTED'
  | 'EXPIRED'
  | 'FAILED';

/**
 * Invitation action type
 */
export type InvitationAction = 
  | 'invite_partner'
  | 'remove_partner';

/**
 * Target entity type
 */
export type TargetEntityType = 'workspace' | 'team';

/**
 * Partner invitation request
 */
export interface InvitePartnerRequest {
  workspaceId: string;
  email: string;
  roleId?: string;
}

/**
 * Partner invitation result
 */
export interface InvitePartnerResult {
  success: boolean;
  email: string;
  status?: InvitationStatus;
  invitationLink?: string | null;
  userId?: string | null;
  roleDisplayName?: string;
  error?: string;
}

/**
 * Remove partner request
 */
export interface RemovePartnerRequest {
  workspaceId: string;
  userId: string;
}

/**
 * Remove partner result
 */
export interface RemovePartnerResult {
  success: boolean;
  userId: string;
  status?: string;
  error?: string;
}

/**
 * Remove partner from team request
 */
export interface RemovePartnerFromTeamRequest {
  teamId: string;
  userId: string;
}

/**
 * Invitation data stored in memory
 */
export interface InvitationData {
  status: InvitationStatus;
  invitationLink: string | null;
  userId: string | null;
  roleDisplayName?: string;
}

/**
 * Invitation link info
 */
export interface InvitationLink {
  email: string;
  invitationLink: string;
  status: InvitationStatus;
}

/**
 * Batch invite partners result
 */
export interface BatchInviteResult {
  success: InvitePartnerResult[];
  failed: Array<{
    email: string;
    error: string;
  }>;
}

/**
 * Batch remove partners result
 */
export interface BatchRemoveResult {
  success: RemovePartnerResult[];
  failed: Array<{
    userId: string;
    error: string;
  }>;
}
